### PR TITLE
[wasm] Allow scalar valuetypes with r4/r8 members.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
Re: https://github.com/dotnet/runtime/issues/94895

File: src/mono/mono/mini/interp/interp.c [hunk 1]
```diff
@@ -1466,9 +1466,13 @@ build_args_from_sig (MonoMethodSignature *sig, InterpFrame *frame)
 
 #ifdef HOST_WASM
 			{
+				MonoType *etype;
+
 				/* Scalar vtypes are passed by value */
-				if (mini_wasm_is_scalar_vtype (sig->params [i]))
+				// FIXME: r4/r8
+				if (mini_wasm_is_scalar_vtype (sig->params [i], &etype) && etype->type != MONO_TYPE_R4 && etype->type != MONO_TYPE_R8) {
 					margs->iargs [int_i] = *(gpointer*)margs->iargs [int_i];
```
Comment (kg) [inline src/mono/mono/mini/interp/interp.c:1474]: Will this work correctly for i8/u8? I'm fine with not fixing those yet though

**Comment by kg** [comment]
LGTM

File: src/mono/mono/mini/interp/interp.c (hunk 1 above)
Comment (vargaz) [reply src/mono/mono/mini/interp/interp.c:1474]: We disable those:
		} else if (size == 8 && t->type != MONO_TYPE_R8) {
			return FALSE;

**Comment by vargaz** [comment]
The Build browser-wasm linux Release LibraryTests_AOT failures are relevant.

**Comment by vargaz** [comment]
Failures are unrelated.

**Review by lewing — APPROVED** [review]

**Pull request merged**